### PR TITLE
Improve Happiness System Summary

### PR DESCRIPTION
**Pull request by gattsuru**
As far as I can tell from the code:
- adds discussion of what happiness does.
- static happiness values are checked when they are evaluated (whether by the Citizen/TownHall gui, by new citizen creation, or by studying tasks), and don't seem to average over past periods (and in fact don't have times set up to average).
- Add's RavenBuilder's rebalance of Saturation happiness.
- citizen-murder doesn't currently detect when the cause of death is a player, rather than some other cause, and notes that guard deaths don't count.
- Adds information about time-based unhappiness and the Social happiness modifier.

This is a little less exact in terms of math, but the numbers used reflect what's required to get a blue or green happiness value.  Some of them are hard to discuss more exactly without becoming confusing, especially death- and damage-related happiness (rather than a negative number, they add a new type of happiness that always returns zero, which means their exact results will change depending on how many other expiration or time-based effects are active).

Merge please :D

### CODE REVIEW

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:13]:
```suggestion
There is an **overall** colony happiness and an **individual** citizen happiness. Higher colony happiness increases the initial level and skills of new colonists. Higher citizen happiness increases the rate colonists gain Intelligence when studying at a [Library](../../source/buildings/library).
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:17]:
```suggestion
If a colonist's saturation above is 7 shanks, they will be happier.
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:18]:
```suggestion
If a colonist's home (or for [Guards](../../source/workers/guard) or trainees, their workplace) is above level 3, the colonist will be happier.
```
Isn't this level 3? And since there are also Knights in Training, maybe remove the link entirely?

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:19]:
```suggestion
If at least two Guards are available for every three workers, the colonists will be happier.
```
If you already linked to the guard page in the last sentence you don't need to link it again

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:21]:
```suggestion
These basic factors benefit from partial completion. Citizens will be happier with 5 shanks of saturation than 1, with 1 Guard for every 4 workers than 1 Guard for every 10, and enjoy a level 2 house more than a level 1 house. Colonists also become happier as these values increase further, such as from upgrading a house to level 4 or 5 or having completely full saturation.
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:23]:
```suggestion
Each time a citizen (other than a Guard) dies, other citizens will mourn the next day and be less happy for the next three days. All citizens become slightly less happy for the next day when injured.
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:27]:
```suggestion
Surviving a raid without losing any colonists provides a colony-wide happiness boost.
```
Does this include Guards dying? If no -> "without losing any citizens"

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:25]:
```suggestion
Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Injuries will reduce colonist happiness, as will Guards who are unable to sleep. Colonists will also become less happy if fellow colonists become sick, starving, homeless, or unemployed (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).
```
"Guards who are unable to sleep" don't they automatically sleep? Why would they not be sleeping?

**Review by ravenbuilder934 — CHANGES_REQUESTED** [review]
> Add's RavenBuilder's rebalance of Saturation happiness.

This'll need to wait until that PR is merged then, I also have some other suggestions though

File: source/systems/happinessandsaturation.md
Comment (gattsuru) [reply source/systems/happinessandsaturation.md:18]: The mechanical rule is BuildingLevel / 2.5, so they should return blue happiness at 2.5 and green happiness at level 2.75 or 3.  The distinction mostly matters for the Barracks, since it's the only building that can have a partial building level.

File: source/systems/happinessandsaturation.md
Comment (gattsuru) [reply source/systems/happinessandsaturation.md:27]: raidManager.onLostCitizen() fires for all citizens, both guards and non-guards, and that's what controls the raidswithoutdeath bonus.  It's only death happiness modifier and need to mourn that skip guard deaths.

File: source/systems/happinessandsaturation.md
Comment (gattsuru) [reply source/systems/happinessandsaturation.md:25]: Ah, think I misunderstood the behavior on that one.  Looks like guards are always going to have slept-tonight fulfilled (or at least at 1.0), while non-guard citizens are affected by sleeping in beds.

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:25]:
```suggestion
Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Injuries will also reduce colonist happiness, as will sleeplessness. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).
```
Maybe this then?

Comment (gattsuru) [reply source/systems/happinessandsaturation.md]: I think I've got injuries twice, so pulled it from there and left the one that's next to deaths.  I tried a different wording that emphasizes that it's specifically inability to get to a bed to sleep that causes the unhappiness :
`Citizens (other than guards) unable to make it to their bed at night to sleep will become unhappy. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).`

Does that look better?

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:17]:
```suggestion
If a colonist's saturation is above 7 shanks, they will be happier.
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:18]:
```suggestion
If the colonist's home (or for [Guards](../../source/workers/guard) or trainees, their workplace) is above level 2.5, the colonist will be happier.
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:19]:
```suggestion
If at least two Guards exist for every three citizens, the colonists will be happier.
```

File: source/systems/happinessandsaturation.md
Comment (ravenbuilder934) [inline source/systems/happinessandsaturation.md:25]:
```suggestion
Every night, colonists will also become less happy if they are sick, homeless, unemployed, or have nothing to do at their job, and will become increasingly unhappy the longer the problem persists. Citizens (other than Guards) unable to make it to their bed at night to sleep will become unhappy. Fellow colonists' livelihood also matters for happiness (the Social Factor in the [Town Hall](../../source/buildings/townhall)'s happiness page).
```

**Review by ravenbuilder934 — COMMENTED** [review]
Closer, but I still have a few nitpicks

**Review by ravenbuilder934 — APPROVED** [review]

**Pull request merged**
